.github/workflows/test.yml

```diff
@@ -4,20 +4,46 @@ jobs:
   test:
     strategy:
       matrix:
-        go-version: [1.14.x, 1.15.x]
-        platform: [ubuntu-latest, windows-latest]
+        go-version: [1.14.x, 1.15.x, 1.16.x]
+        platform: [ubuntu-20.04, windows-latest]
     runs-on: ${{ matrix.platform }}
     steps:
     - name: Install Go
-      uses: actions/setup-go@v1
+      uses: actions/setup-go@v2
       with:
         go-version: ${{ matrix.go-version }}
     - name: Checkout code
       uses: actions/checkout@v2
-    - name: Test
-      run: make test
     - name: Lint
       run: make lint
     - name: Cross build
       if: ${{ runner.os == 'Linux' }}
       run: make cross
+    - name: Test (!linux)
+      if: ${{ runner.os != 'Linux' }}
+      run: make test
+    - name: Test (linux)
+      if: ${{ runner.os == 'Linux' }}
+      run: |
+        sudo -E PATH=$PATH make test
+        # fixup go cache screwed by sudo
+        sudo chown -R $(id -u):$(id -g) $HOME/.cache
```

Comment:
Some questions; 

- Is the cache actually used (or "useful"), assuming these nodes are cleaned between runs?
- Did this cause a problem in CI, or only on your local machine (testing `sudo -E PATH=... `, after which the files were owned by `root`)? (if so, it may not be needed in CI, as state would be cleared after this completes, correct?)
- Alternatively, we can set `GOCACHE` so that the `sudo` variant writes to it's own home directory, not the home directory of the non-privileged user, e.g.

    ```bash
    sudo -E PATH=$PATH GOCACHE=/root/.cache make test
    ```

Reply:
> Is the cache actually used (or "useful"), assuming these nodes are cleaned between runs?

It is used (based by the errors in CI I saw), and I guess it is mildly useful. Surely, we're not compiling docker here, so the savings are not that big.

I am fixing the .cache ownership not because the cache is useful, but for the sake of future contributors who might add some commands after that one, and discover that it fails.

Actually, the proper way would be to compile the tests as a user and run them as root, but I'm not sure how to do it in an elegant way.

> Did this cause a problem in CI

Yes it did, as "make lint" and "make cross" were before "sudo make test" and they failed because they could not write to `~/.cache`. So I mitigated it with two things -- (1) moved all non-sudo commands to before sudo ones (2) added this fixup.

> we can set `GOCACHE`

Good idea, let me try. Judging by the amount of code to compile (our packages plus a bits of x/sys/unix), caching should not result in any dramatic improvements.

Reply:
Ended up implementing decent `make sudotest` via Makefile, which compiles tests as a user and runs them as root. This way we don't have to care about gocache and other similar things (like re-downloading modules).

Reply:
Scratch that bloody sudotest, I found a _much_ better way, see the last commit.